tests/test_core.py

```diff
@@ -38,19 +39,106 @@ class test_TypeDict(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different list types."""
-        raise NotImplementedError()
+
```

Comment:
Don't leave a whitespace between docstrings and code. (You can delete here and below.)